### ISSUE TITLE
implement 'relay only' mode

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#4197]
It would be very useful to have a mode where you can run ipfs entirely over relayed connections. This would allow lower powered devices to be able to interact with the ipfs network more easily.

**Comment by kevina** [comment]
I agree, I think I may of brought this up before, although it might of only been on IRC.  It has lots of uses beyond a power usage.

**Comment by Stebalien** [comment]
Related but slightly different (for people reading one but not the other): #4110

**Comment by magik6k** [comment]
Some sort of RemoteBlockstore which would pass block requests to a Gateway is an option too. With API Tokens this could be extended to some sort of light node which would manage it's own secrets, but rely on other peers for connectivity.

**Comment by whyrusleeping** [comment]
@magik6k oooh, yeah. that would be really nice, passing getblocks calls to `ipfs.io/api/v0/blocks/get`

**Comment by vyzo** [comment]
Of course such nodes will still need to bootstrap using regular connectivity.

**Comment by whyrusleeping** [comment]
We could probably implement a very simple version of this by allowing users to specify no swarm listener addresses. Then if they dial out to a relay peer, others can connect to them via the relay.

**Comment by vyzo** [comment]
We will also need to advertise the relay addresses to the DHT in this case.

**Comment by whyrusleeping** [comment]
@vyzo Yeah, good point. We can probably do that by putting them in the Addresses.Announce config field.

**Comment by Foxcool** [comment]
Also this ligth mode can be used on laptops.

Linux laptop with SYSTEMD or another service manager. On AC-power all p2p nodes used on full mode: download, upload. But on battery mode all services reloads in light mode and only read the data.
